infrastructure/infrastructure.py

```
@@ -0,0 +1,476 @@
+    instances = [manager, worker_0, worker_1]
+
+    print("Setting up infrastructure...")
+
+    with ThreadPoolExecutor(max_workers=3) as executor:
+        futures = [executor.submit(setup_instance_chain, instance) for instance in instances]
+        results = []
+        for instance, future in zip(instances, futures, strict=False):
+            try:
+                result = future.result()
+                results.append(result)
+            except Exception as e:
```

Ruff (0.13.0)

Do not catch blind exception: `Exception`

(BLE001)